src/agent0/ethpy/base/transactions_test.py

```diff
@@ -0,0 +1,132 @@
+"""Test transaction behavior."""
+
+import logging
+
+import pytest
+from fixedpointmath import FixedPoint
+from hexbytes import HexBytes
+
+from agent0.core.hyperdrive.interactive import ILocalChain, ILocalHyperdrive
+from agent0.ethpy.base.transactions import build_transaction
+
+# we're testing here
+# pylint: disable=too-many-locals
+
+
+@pytest.mark.anvil
+def test_gas_price_base_multiple_explicit(chain: ILocalChain):
+    """Set the gas price base multiple explicitly."""
+    # set up config, hyperdrive, interface, web3, and agent
+    base_fee_multiple = 100
+    regular_config = ILocalHyperdrive.Config()
+    regular_hyperdrive = ILocalHyperdrive(chain, regular_config)
+    regular_interface = regular_hyperdrive.interface
+    regular_web3 = regular_interface.web3
+    regular_agent = regular_hyperdrive.init_agent(eth=FixedPoint(1))
+
+    fn_args_mint = (
+        regular_agent.checksum_address,  # destination
+        FixedPoint(11111).scaled_value,  # amount
+    )
+    regular_built_transaction = build_transaction(
+        func_handle=regular_interface.base_token_contract.functions.mint(*fn_args_mint),
+        signer=regular_agent.agent,
+        web3=regular_web3,
+    )
+    assert "maxPriorityFeePerGas" in regular_built_transaction
+    assert isinstance(regular_built_transaction["maxPriorityFeePerGas"], int)
+    regular_priority_fee_per_gas = regular_built_transaction["maxPriorityFeePerGas"]
+    assert "maxFeePerGas" in regular_built_transaction
+    assert isinstance(regular_built_transaction["maxFeePerGas"], int)
+    regular_base_fee_per_gas = regular_built_transaction["maxFeePerGas"] - regular_priority_fee_per_gas
+
+    multiplied_built_transaction = build_transaction(
+        func_handle=regular_interface.base_token_contract.functions.mint(*fn_args_mint),
+        signer=regular_agent.agent,
+        web3=regular_web3,
+        txn_options_base_fee_multiple=base_fee_multiple,
+    )
+    assert "maxPriorityFeePerGas" in multiplied_built_transaction
+    assert isinstance(multiplied_built_transaction["maxPriorityFeePerGas"], int)
+    multiplied_priority_fee_per_gas = multiplied_built_transaction["maxPriorityFeePerGas"]
+    assert "maxFeePerGas" in multiplied_built_transaction
+    assert isinstance(multiplied_built_transaction["maxFeePerGas"], int)
+    multiplied_base_fee_per_gas = multiplied_built_transaction["maxFeePerGas"] - multiplied_priority_fee_per_gas
+
+    assert multiplied_base_fee_per_gas / regular_base_fee_per_gas == base_fee_multiple
+
+
+@pytest.mark.anvil
+def test_gas_price_priority_multiple_explicit(chain: ILocalChain):
```

Comment:
I'm not seeing the difference between this test and the one above it

Reply:
the one above it tests the base fee multiple, this tests the priority fee multiple

Reply:
Ahhh got it, thanks!